### PR TITLE
Fix konflux index and bundle cel expressions to avoid build loops

### PR DESCRIPTION
**Pull request by pierDipi**
Fix the bundle + index build looping

Change index directory from being within bundle manifests to be it's own under `olm-catalog/serverless-operator-index`

I will port this changes to the hack repo once merged

### CODE REVIEW

File: .tekton/serverless-ingress-135-pull-request.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ metadata:
     build.appstudio.redhat.com/target_branch: '{{target_branch}}'
     pipelinesascode.tekton.dev/max-keep-runs: "3"
     pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
-      == "main" 
+      == "main" && ( files.all.exists(x, !x.matches('^olm-catalog/')) || files.all.exists(x, x.matches('^.tekton/')) )
```
Comment (creydr) [inline .tekton/serverless-ingress-135-pull-request.yaml:11]: Not sure if we need to have the `files.all.exists(x, x.matches('^.tekton/')`, as as long we don't have only changes in `olm-catalog/` we can build the ingress.
Probably similar in the other components.

https://playcel.undistro.io/?content=H4sIAAAAAAAAA3WSUWvCMBDHv8rZPVRB67tvMoRtuE02hgxkcLZnG5bkSpJqZey779J1Ux8kD73e5f6%2F%2FyX5SnLSySyZTmFNOmdDEBhCRXC7WMJK47F03NhisLGy5TIHygNaUDaQwzyoPYnGdu49ma0%2BQs0HclQA2b1ybA3ZELWprTU7ks4ixuRUVzmoUP1y2Ri2sGhrR94rCZdoywZLgqHwR1l00rmZwNqpQHDkxnXW6NRTCbrf9OapE0ZHCFKKsVcFwY6dmK%2BbAAUGHEsM7%2FPHJUj64fX5KdYNhl5lFZUhfWls2k2xR93gFTiWqKwPHekE6IUW%2FfyxL2etSU5OengnCmhqTb435mvlMNY2snZKChlqnVGrfPDDdgyDNhN%2FeUV%2BmH6wNpNcKJrLaToaJeMkMu8jXa73Bu7kQOKFXXq6MnP2j5xtLIBguy%2BI%2FXPOXy4L9BnYyr9gDRckxPisvn8A84xfTF4CAAA%3D

Comment (pierDipi) [reply .tekton/serverless-ingress-135-pull-request.yaml:11]: agree, I will fix it on the hack PR as this change is mostly to make it work, ok?

Comment (creydr) [reply .tekton/serverless-ingress-135-pull-request.yaml:11]: /lgtm

**Comment by creydr** [comment]
```
ERRO[2024-10-08T07:45:31Z] 
  * could not run steps: step serverless-index failed: error occurred handling build serverless-index-amd64: the build serverless-index-amd64 failed after 13m7s with reason ManageDockerfileFailed: Failed to prepare the dockerfile for the build. 
```
probably as we need to regenerate the ci config with the new paths

**Comment by pierDipi** [comment]
Yes, hack repo change incoming

**Comment by pierDipi** [comment]
https://github.com/openshift-knative/hack/pull/320 that will fix CI

**Pull request merged**